netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -255,7 +242,13 @@ private NettyClientHandler newHandler() {
     frameWriter = new Http2OutboundFrameLogger(frameWriter, frameLogger);
 
     BufferingHttp2ConnectionEncoder encoder = new BufferingHttp2ConnectionEncoder(
-            new DefaultHttp2ConnectionEncoder(connection, frameWriter));
+            new DefaultHttp2ConnectionEncoder(connection, frameWriter)) {
+      @Override
+      public ChannelFuture writeSettingsAck(ChannelHandlerContext ctx, ChannelPromise promise) {
+        listener.transportReady();
```

Comment:
We could receive many acks over the course of a connection. We should keep a boolean of whether or not we are already ready (so we don't call `transportReady()` multiple times).